src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/DoNotDecreaseInheritedMemberVisibilityTests.Fixer.cs

```diff
@@ -12,22 +11,387 @@ public class DoNotDecreaseInheritedMemberVisibilityFixerTests : CodeFixTestBase
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
         }
 
         protected override CodeFixProvider GetBasicCodeFixProvider()
         {
-            return new BasicDoNotDecreaseInheritedMemberVisibilityFixer();
+            return new DoNotDecreaseInheritedMemberVisibilityFixer();
         }
 
         protected override CodeFixProvider GetCSharpCodeFixProvider()
         {
-            return new CSharpDoNotDecreaseInheritedMemberVisibilityFixer();
+            return new DoNotDecreaseInheritedMemberVisibilityFixer();
+        }
+
+        [Fact]
+        public void DecreaseCSharpMemberVisibility()
+        {
+            VerifyCSharpFix(
+// Original Code                
+@"public class BaseClass
+{
+    public int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    private new int MyProp3;
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    public int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3;
+}");
+
+            VerifyCSharpFix(
+// Original Code
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    internal new int MyProp3 { get; set; }
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 { get; set; }
+}"
+);
+
+            VerifyCSharpFix(
+// Original Code
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 { get; private set; }
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 { get; set; }
+}");
+
+            VerifyCSharpFix(
+// Original Code
+@"public class BaseClass
+{
+    public void MyProp3() {}
```

Comment:
Yea, I was a little sloppy with copy/pasting names here. I'll fix them up to sensible names based on the types of members they apply to.